reactnative/RNPlayground/app/screens/PlaygroundScreen.tsx

```diff
@@ -1,76 +1,108 @@
+}
 
-export const PlaygroundScreen: FC<StackScreenProps<AppStackScreenProps, "Playground">> = observer(function PlaygroundScreen() {
-    const $bottomContainerInsets = useSafeAreaInsetsStyle(["bottom"])
+export const PlaygroundScreen: React.FC<PlaygroundScreenProps> = observer((props) => {
+  const $bottomContainerInsets = useSafeAreaInsetsStyle(["bottom"])
+  const { playgroundStore } = useStores()
 
-    return (
-       <View style={$container}>
-         <View style={$topContainer}>
-           <Text
-             testID="welcome-heading"
-             style={$welcomeHeading}
-             tx="playgroundScreen.readyForLaunch"
-             preset="heading"
-           />
-           <Text tx="playgroundScreen.postscript" preset="subheading" />
-         </View>
+  useFocusEffect(
+    React.useCallback(() => {
+      playgroundStore.startCounter()
+      return () => playgroundStore.stopCounter()
+    }, [playgroundStore])
+  )
 
-         <View style={[$bottomContainer, $bottomContainerInsets]}>
-           <Text tx="playgroundScreen.exciting" size="md" />
-         </View>
-       </View>
-     )
- })
+  return <View style={$container}>
+    {
+      StateComponent(
+        $bottomContainerInsets,
+        playgroundStore.uiState,
+        () => playgroundStore.resetCounter()
+      )
+    }
+  </View>
+})
 
- const $container: ViewStyle = {
-   flex: 1,
-   backgroundColor: colors.background,
- }
+function StateComponent(insets, uiState, onResetClicked) {
+  const { loading, error, content } = uiState
+  if (loading) {
+    return LoadingComponent(insets)
+  } else if (error) {
+    return ErrorComponent(insets, error, onResetClicked)
+  } else if (content) {
+    return ContentComponent(insets, content, onResetClicked)
+  }
+  throw new Error(`UiState type is not handled ${JSON.stringify(uiState)}`)
+}
 
- const $topContainer: ViewStyle = {
-   flexShrink: 1,
-   flexGrow: 1,
-   flexBasis: "57%",
-   justifyContent: "center",
-   paddingHorizontal: spacing.large,
- }
+function ContentComponent(insets, content, onResetClicked) {
+  return <View>
+    <View style={$topContainer}>
+      <Text
+        testID="welcome-heading"
+        style={$welcomeHeading}
+        tx="playgroundScreen.readyForLaunch"
+        preset="heading"
+      />
+      <Text tx="playgroundScreen.postscript" preset="subheading" />
+    </View>
 
- const $bottomContainer: ViewStyle = {
-   flexShrink: 1,
-   flexGrow: 0,
-   flexBasis: "43%",
-   backgroundColor: colors.palette.neutral100,
-   borderTopLeftRadius: 16,
-   borderTopRightRadius: 16,
-   paddingHorizontal: spacing.large,
-   justifyContent: "space-around",
- }
- const $welcomeLogo: ImageStyle = {
-   height: 88,
-   width: "100%",
-   marginBottom: spacing.huge,
- }
+    <View style={[$bottomContainer, insets]}>
+      <Text tx="playgroundScreen.exciting" size="md" />
+      <Text text={`Value is: ${content.number}`} size="md" />
```

Comment:
@juankysoriano  had a question in the nav PR I didn't get to - but this shows off the difference between `tx=screen.stringName` and `text={String Literal}`. Hope this helps